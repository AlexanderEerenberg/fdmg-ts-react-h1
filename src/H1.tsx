import * as React from 'react';

/**
 * H1 compononent renders a heading
 */
export default class H1 extends React.Component {
    public state: any;
    public props: any;

    constructor(props: any) {
        super(props);
        this.props = props;
        console.info('H1 component', props);
    }

    componentDidMount() {
        console.info('H1 mounted');
    }
        
    componentWillUnmount() {
        console.info('H1 will unmount');
    }

    componentWillReceiveProps(nextProps) {
        console.info('H1 next props', nextProps);
    }

    render() {
        return (
            <h1 className={this.props.className}>{this.props.text}</h1>
        );
    }
}
