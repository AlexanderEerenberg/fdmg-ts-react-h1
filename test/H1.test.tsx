import * as React from 'react';
import * as renderer from 'react-test-renderer';
import H1 from '../src/H1';

console.info = function() {};

test('H1 renders correctly', () => {
    const component = renderer.create(
        <H1 text={'test h1'}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('H1 renders correctly with CSS class', () => {
    const component = renderer.create(
        <H1 text={'test h1'} className={'title'}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});