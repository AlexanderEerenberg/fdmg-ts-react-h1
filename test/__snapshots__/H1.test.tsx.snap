// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 renders correctly 1`] = `
<h1
  className={undefined}
>
  test h1
</h1>
`;

exports[`H1 renders correctly with CSS class 1`] = `
<h1
  className="title"
>
  test h1
</h1>
`;
